File: Python/Kode Program/Customer.py
from Items import Items

class Customer:
    def __init__(self, Nama="", Tanggal=""):
        self.__Nama = Nama
        self.__Tanggal = Tanggal
        self.__pesanan = []

    def tambah_pesanan(self, item):
        if isinstance(item, Items):
            self.__pesanan.append(item)
        else:
            print("Pesanan harus berupa objek dari kelas Items atau turunannya.")

    def set_nama(self, nama):
        self.__Nama = nama

    def get_nama(self):
        return self.__Nama

    def set_tanggal(self, tanggal):
        self.__Tanggal = tanggal

    def get_tanggal(self):
        return self.__Tanggal

    def get_pesanan(self):
        return self.__pesanan

File: CPP/Kode Program/Customer.cpp
#ifndef CUSTOMER_CPP
#define CUSTOMER_CPP

#include <iostream>
#include <string>
#include <vector>
#include "Items.cpp"

using namespace std;

class Customer
{
    private:
        string Nama;                         // Nama Customer
        string Tanggal;                      // Tanggal Pemesanan
        vector<Items*> pesanan;              // List pesanan (menggunakan pointer ke Items)
    
    public:
        // Constructor Default
        Customer() 
        {

        }

        // Setter Nama
        void setNama(string nama1)
        {
            Nama = nama1;
        }

        // Getter Nama
        string getNama()
        {
            return Nama;
        }

        // Setter Tanggal
        void setTanggal(string tanggal1)
        {
            Tanggal = tanggal1;
        }

        // Getter Tanggal
        string getTanggal()
        {
            return Tanggal;
        }

        // Menambahkan Pesanan ke dalam daftar
        void tambahPesanan(Items* item)
        {
            pesanan.push_back(item);
        }

        // Menampilkan Daftar Pesanan
        void tampilkanPesanan()
        {
            cout << "Daftar Pesanan untuk Customer: " << Nama << endl;
            for (Items* item : pesanan)
            {
                cout << "- " << item->getNama() << " (Rp" << item->getHargaProduk() << ")" << endl;
            }
        }

        // Destructor untuk menghapus semua pesanan (Pointer Cleanup)
        ~Customer()
        {
            for (Items* item : pesanan)
            {
                delete item;  // Menghapus objek yang di-pointer agar tidak terjadi memory leak
            }
            pesanan.clear();
        }
};

#endif  // CUSTOMER_CPP